Revamping AboutViewController

jleandroperez opened the pull request:
In this PR we implement the following features:

- AboutViewController's UI has been entirely rewritten
- The *About* row has been moved from *SupportViewController* over to *MeViewController*, now it's top level.
- The App Version was being displayed as: `Version (Build)` (plus `(Dev)` when applicable. This has been simplified to just `Version` (thanks @sendhil for the feedback there!)

Props to @drw158 for providing the new and awesome design!

Closes #3998

Needs Review: @aerych 

--

#### Scenario 1: MeViewController with Twitter Row
1. Make sure that the Twitter App is deleted from your device
2. Fresh install WPiOS
3. Log into your account
4. Tap over `Me` > `About`

As a result, About should be onscreen, with the *Twitter* row visible.

-- 

#### Scenario 2: MeViewController without Twitter Row
1. Make sure that the Twitter App is not installed in your device
2. Fresh install WPiOS
3. Log into your account
4. Tap over `Me` > `About`

As a result, the About should be onscreen, but the Twitter row should be missing in action.

jleandroperez commented:
#### iPhone 5s Screenshot:

![ios simulator screen shot jul 9 2015 8 14 49 pm](https://cloud.githubusercontent.com/assets/1195260/8609032/4aa0c480-2677-11e5-8a2c-fd4c4cdedb3d.png)

jleandroperez commented:
#### iPad Air Screenshot:

![ios simulator screen shot jul 9 2015 8 16 55 pm](https://cloud.githubusercontent.com/assets/1195260/8609057/7b9877d6-2677-11e5-910b-2d39ab1843f5.png)

drw158 commented:
Do we need the stuff in parentheses? 

![screen shot 2015-07-09 at 6 27 09 pm](https://cloud.githubusercontent.com/assets/618551/8609217/2d0ed29a-2668-11e5-8fb3-1bd0dda87cd0.png)

Twitter username should be capitalized `@WordPressiOS`

There should be a period at the end of `Inc` : (`Automattic, Inc.`)

jleandroperez commented:
@drw158 Strings updated!. As per the number in parentheses, those fields are actually coming from different spots. In our specific case, they match in the Dev / Internal / AppStore builds, but they could differ.

@diegoreymendez do you think it'd be wise to just render the `CFBundleShortVersionString` string? That one comes from the `Version` project field, while `CFBundleVersion` loads the `Build` project field.

[Reference here](http://stackoverflow.com/questions/19726988/what-should-i-put-for-cfbundleversion-and-cfbundleshortversionstring/19728342#19728342)

diegoreymendez commented:
@jleandroperez - I'd ask @sendhil if there's any plan to have different numbers there.  The more detailed version number would be preferred imho.  In any case I'd say there's no need to have two numbers.

drw158 commented:
@jleandroperez Ah ok, I asked because both of the numbers seem to always be the same, and it was redundant. It's fine if we have both, but it would be preferred If we only show the short version string to users. I wasn't sure if we showed both only in the internal build.

jleandroperez commented:
Thanks for the feedback @diegoreymendez!. I'll poke Sendhil, and if possible, i'll make the arrangements so we only show one string.

Thanks for checking @drw158!!

jleandroperez commented:
#### Update:

Sendhil's confirmed that it'd be okay to keep just the Version string. Refreshed screenshot here:

![ios simulator screen shot jul 13 2015 10 08 56 am](https://cloud.githubusercontent.com/assets/1195260/8650172/36de8522-2947-11e5-8cc1-99e30d469c62.png)

drw158 commented:
One final nitpick: I think `Rate Us on the App Store` should be changed to `Rate us on the App Store`.

`Us` has been de-capitalized since that text has a sentence structure. We haven't standardized capitalization in table rows yet, but I think this is the way we are going.

jleandroperez commented:
@drw158 string updated, and ready for review. Thanks Dave!

aerych commented on WordPress/Classes/ViewRelated/Settings/AboutViewController.swift:
```diff
@@ -0,0 +1,230 @@
+    // MARK: - Private Properties
+    private var footerView : WPTableViewSectionFooterView!
+    
+    private var rowTitles : [[String]] {
```
I'm not sure having three methods to describe the row data, each building its own array, is the best approach. Would it make more sense to declare a type with the relevant properties and have a single array?  Seems like this would be more straight forward to update in the future.

aerych commented:
Hi @jleandroperez!  

I have a couple of observations from testing:
#### Padding
In landscape orientation there isn't much padding at the bottom.
![ios simulator screen shot jul 13 2015 1 59 05 pm](https://cloud.githubusercontent.com/assets/1435271/8658902/733f9a40-296c-11e5-9ad1-62bb56a0a398.png)

The other screens have a bit more padding. Maybe we could add some space on the about screen for consistency.  For example:
![ios simulator screen shot jul 13 2015 1 59 02 pm](https://cloud.githubusercontent.com/assets/1435271/8658916/8ca43b4e-296c-11e5-9402-ae0ef1a75c0a.png)

#### Version
Since we're showing the app version in the About screen, should we remove it from the Help screen? 
![ios simulator screen shot jul 13 2015 2 26 16 pm](https://cloud.githubusercontent.com/assets/1435271/8658933/b95e3ef0-296c-11e5-9207-cb299c582044.png)

#### Modal
On the iPad, if I tap the help button on the sign in screen then the Help & Support screen appears as a modal, and if I tap About, the about screen is also shown in the same modal. This makes sense.  On the other hand, if I'm logged in and tap Me > Help & Support this screen is not modal. Tapping About still launches the About screen in a modal window.  Would it make sense to push this on to the navigation stack full screen instead? The modal feels a bit weird to me in this context. 
![ios simulator screen shot jul 13 2015 2 41 46 pm](https://cloud.githubusercontent.com/assets/1435271/8659014/4e4862fc-296d-11e5-8c67-9a28bee3566e.png)

Related, notice that the web pages opened from the About vc have horizontal scrolling when its in a modal state:
![ios simulator screen shot jul 13 2015 2 24 36 pm](https://cloud.githubusercontent.com/assets/1435271/8659041/7f417268-296d-11e5-8c44-6a40dc7c04c1.png)


#### WebViewVC Titles
This isn't necessarily related to this PR, but notice on the iPhone that the title and url shown in the navigation bar is off center.  I believe this is because we're showing the About page's title.  IIRC other places we display the web view controller we make sure to set the previous vc navigation item's back item to an empty string to prevent this.  Should we do the same here, or would it be better to iterate on  WPWebViewController to make sure its title + url is visually centered within the parent view, not the custom title view. 
![ios simulator screen shot jul 13 2015 1 57 00 pm](https://cloud.githubusercontent.com/assets/1435271/8659047/8c3ef576-296d-11e5-918a-d61255a9224d.png)

#### Web Page appearance
I noticed the menu of the app's blog looks very odd on mobile. Out of scope for this PR but just wanted to note it here.  cc @drw158 because I think you're familiar with the apps page design. 
![ios simulator screen shot jul 13 2015 1 55 23 pm](https://cloud.githubusercontent.com/assets/1435271/8659131/244d9930-296e-11e5-929e-d7865bcaa5cb.png)

#### WordPress.org vs Automattic
Its a bit odd to see the blog titled "WordPress.org Mobile Apps" and then see a copyright Automattic Inc. on the About screen.

jleandroperez commented:
Impressive feedback @aerych!! thanks a lot!. I've already addressed the following issues:

- Implemented *Row* Nested class
- Added extra bottom padding in the AboutViewController's UITableView
- WebViewVC: Fixed the "Back" NavBar UI glitch

As per the following, i'd defer to @drw158's opinion on what to do:
- Should we nuke the Version from the Support View?
- `Webpage Appearance` and `WordPress.org vs Automattic`: Outside the scope of this PR. Maybe we shouldn't link the Mobile Blog?

drw158 commented:
> Should we nuke the Version from the Support View?

There was a version number in both screens before as well, but now it's more apparent since they have the same styling. I think it's helpful to have the version number right there in support.

This made me think... does it make more sense for About to be one level higher? (with Account Settings and Help & Support)

> Webpage Appearance and WordPress.org vs Automattic: Outside the scope of this PR. Maybe we shouldn't link the Mobile Blog?

Let's keep it for now. Unfortunately we have to keep the blog .org branded.


I'm currently looking at the modal stuff on iPad and I think I agree that a push rather than a modal would work better there. Honestly, I forgot why I thought modal was more appropriate...

jleandroperez commented:
@drw158 all changes in place!. Should we move the About View one level up, and display it as a row below `Help and Support`?

drw158 commented:
@jleandroperez I think that would be a good idea. In order to emulate the Me view on the web, can we add a space right below Account Settings so that it's separate from the rest of the rows? This will separate profile related items from "other" menu items.

![screen shot 2015-07-14 at 10 03 27 am](https://cloud.githubusercontent.com/assets/618551/8676561/f2af53d0-2a0f-11e5-9fdc-c45c555768af.png)

jleandroperez commented:
#### Update:
*About* is now a top level row, available in the `Me` tab. Sample screenshots below (iPhone 5 + iPhone 6 screen sizes + iPad).

Plus, it's not modal anymore, just being pushed into the navigation stack. @drw158 does this look okay to you?.

Thanks!


![ios simulator screen shot jul 14 2015 3 04 38 pm](https://cloud.githubusercontent.com/assets/1195260/8681097/dc888dca-2a3b-11e5-86b6-133e37bd8eb1.png)
![ios simulator screen shot jul 14 2015 3 05 16 pm](https://cloud.githubusercontent.com/assets/1195260/8681098/dcdff36c-2a3b-11e5-9548-b79265e584c8.png)
![ios simulator screen shot jul 14 2015 3 19 30 pm](https://cloud.githubusercontent.com/assets/1195260/8681099/dce132c2-2a3b-11e5-8d6c-6f63ec15114b.png)

jleandroperez commented:
@aerych ready for a second round. Thank you sir!

drw158 commented:
Looks good with the space.

aerych commented:
Looks great @jleandroperez :) 
:shipit:

jleandroperez commented:
@aerych thanks *a lot* for helping us out!!
